test-framework/src/main/java/org/apache/brooklyn/test/framework/TargetableTestComponentImpl.java

```diff
@@ -116,4 +125,14 @@ private static Entity getTargetById(final ExecutionContext executionContext, fin
     protected <T> T getRequiredConfig(ConfigKey<T> key) {
         return checkNotNull(config().get(key), "config %s must not be null", key);
     }
+
+    protected void setUpAndRunState(boolean up, Lifecycle status) {
+        if (up) {
+            sensors().set(SERVICE_UP, up);
+            setExpectedState(this, status);
+        } else {
+            setExpectedState(this, status);
+            sensors().set(SERVICE_UP, up);
+        }
```

Comment:
As you are updating this anyway would be nice to make the `service.isUp` + `service.state` managed through the notUp + problems maps, instead of directly. 
Current approach is valid as well so happy to go this way if you think it's out of scope.